src/uptane/tuf.cc

```diff
@@ -91,12 +91,9 @@ bool Hash::MatchWith(const std::string &content) const {
 
 bool Hash::operator==(const Hash &other) const { return type_ == other.type_ && hash_ == other.hash_; }
 
-Target::Target(const std::string &filename, const Json::Value &content) : filename_(filename), ecu_identifier_("") {
+Target::Target(const std::string &filename, const Json::Value &content) : filename_(filename) {
   if (content.isMember("custom")) {
-    Json::Value custom = content["custom"];
-    if (custom.isMember("ecuIdentifier")) {
-      ecu_identifier_ = content["custom"]["ecuIdentifier"].asString();
-    }
+    custom_ = content["custom"];
```

Comment:
My concern is that returning a Json::Value means that the users of Target are locked-in to the specific serialization format for targets.  If we decide to support ASN.1 DER encoded targets files, then the Json::Value Target::custom() member function will need changing everywhere.

Instead of an explicit property, why not put the logic for matching a target against an EcuIdentifier inside the Target object itself?  Done that was it doesn't matter where the Target object came from.  You just ask 'are you for this EcuIdentifier?'.  Targets from image repository will always say 'yes', but targets from director will check the custom field.

In the future we may want a 'Ecu' object that includes both the ecuIdentifier and hardwareIdentifier.  In that case the Image repository would perform a check against hardwareIdentifier, but the users of a Target object wouldn't need to worry about the details.

An std::string Targets::ecu_identifier() member function would be fine too, but it shouldn't return Json :)